// Copyright 2024 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
syntax = "proto2";

package koladata.internal;

import "koladata/s11n/codec.proto";

// Error for not finding a common schema when creating DataSlice. This is used
// for carrying error information and rendering readable error message to user.
message NoCommonSchema {
  // The most common schema fund in the input.
  optional s11n.KodaV1Proto.DataItemProto common_schema = 1;
  // The conflict schema that cannot be converted to the common schema.
  optional s11n.KodaV1Proto.DataItemProto conflicting_schema = 2;
}

// Koda error message transportation proto.
message Error {
  optional string error_message = 1;
  optional Error cause = 2;
  oneof error {
    NoCommonSchema no_common_schema = 3;
  }
}
